src/core/parser_internal/docgen_worker.hpp

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include "../symbol.hpp"
+#include "../token.hpp"
+#include "../json_writer.hpp"
+#include "../parse_worker.hpp"
+#include "../worker_routine.hpp"
+
+namespace docgen {
+namespace core {
+namespace parser_internal {
+
```

Comment:
add `using symbol_t = Symbol;`. It's still implementation detail that you are using `Symbol` in particular as the "symbol" type.

Reply:
As it pertains to the parser, I'm not using `Symbol` as an implementation detail. I am using the specialization `Token<Symbol>`—as far as my parser is concerned, this is a single normal type. A `symbol_t` alias would only be referenced in the `token_t` declaration, and nowhere else.